### ISSUE TITLE
template-curly-spacing does not always work with babel-eslint

### ISSUE THREAD

**Issue by eventualbuddha** [eslint/eslint#7854]
**Tell us about your environment**

macOS Sierra, iTerm 2.0

* **ESLint Version:** 3.12.2
* **Node Version:** 5.1.0
* **npm Version:** 3.3.12

**What parser (default, Babel-ESLint, etc.) are you using?**

babel-eslint

**Please show your full configuration:**

```js
{
  "parser": "babel-eslint"
  "rules": {
    "template-curly-spacing": ["error", "never"]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

See demo repo [here](https://github.com/eventualbuddha/eslint-babel-template-curly-spacing-issue).

```js
`a${(function() { return `b${ 0 }c`; })}d`;
```

**What did you expect to happen?**

I expect two errors, one on both sides of the `0` inside the nested template literal.

**What actually happened? Please include the actual, raw output from ESLint.**

I got one error:

```
/Users/donovan/Desktop/eslint-babel-template-curly-issue/index.js
  1:33  error  Unexpected space(s) before '}'  template-curly-spacing

✖ 1 problem (1 error, 0 warnings)
```

eslint it not reporting the space after `${` as a problem. This is due to the difference in tokens between espree and babylon. It may actually be a bug in babylon, but I'm not sure. See the [demo repo](https://github.com/eventualbuddha/eslint-babel-template-curly-spacing-issue) for further analysis.

**Comment by vitorbal** [comment]
Thanks for the report. To be clear, are you reporting a bug with babel-eslint or with the behavior with the default parser?

**Comment by eventualbuddha** [comment]
With babel-eslint.

**Comment by platinumazure** [comment]
@eventualbuddha Do you get your expected results with the default parser?

**Comment by eventualbuddha** [comment]
Yes

**Comment by vitorbal** [comment]
@eventualbuddha thanks for the quick feedback!
In that case, I would suggest you post this issue on the [babel-eslint repository](https://github.com/babel/babel-eslint) instead, since they are the ones responsible for maintaining the babel-eslint parser.

**Comment by eventualbuddha** [comment]
Posted as https://github.com/babel/babel-eslint/issues/429, closing this.

**Comment by eventualbuddha** [comment]
One final note, I saw that https://github.com/eslint/eslint/pull/6290 was created to fix an issue with babel-eslint so I decided that it might be worth opening here. Is there guidance somewhere on which repo to open an issue in?

**Comment by vitorbal** [comment]
@eventualbuddha looks like that issue was an exception, as can be seen in the discussion here: https://github.com/eslint/eslint/issues/6266#issuecomment-222560132
As a rule of thumb, if the buggy behavior happens with `babel-eslint` but not with the default parser, then you should report an issue in the `babel-eslint` repo.
